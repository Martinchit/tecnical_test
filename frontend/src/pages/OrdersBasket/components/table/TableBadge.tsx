import React from 'react';
import styled from 'styled-components';
import { Badge as B } from 'react-bootstrap';
import { OrderStatus } from '../../../../types/types';

interface TableBadge {
  orderStatus: OrderStatus;
}

export const TableBadge: React.FC<TableBadge> = ({ orderStatus }) => {
  const getVariant = () => {
    switch (orderStatus) {
      case 'Booked':
        return 'success';
      case 'In Progress':
        return 'info';
      case 'Rejected':
        return 'danger';
      case 'Ready':
        return 'primary';
      default:
        return 'warning';
    }
  };
  return <Badge variant={getVariant()}>{orderStatus}</Badge>;
};

const Badge = styled(B)`
  width: 80%;
  @media (max-width: 767px) {
    width: 100%;
  }
`;
