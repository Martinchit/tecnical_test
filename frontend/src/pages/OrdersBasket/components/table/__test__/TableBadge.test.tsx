import React from 'react';
import { shallow } from 'enzyme';

import { TableBadge } from '../TableBadge';

describe('TableBadge', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<TableBadge orderStatus="Booked" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders success Badge', () => {
    const wrapper = shallow(<TableBadge orderStatus="Booked" />);
    expect(wrapper.props().variant).toBe('success');
  });

  it('renders info Badge', () => {
    const wrapper = shallow(<TableBadge orderStatus="In Progress" />);
    expect(wrapper.props().variant).toBe('info');
  });

  it('renders danger Badge', () => {
    const wrapper = shallow(<TableBadge orderStatus="Rejected" />);
    expect(wrapper.props().variant).toBe('danger');
  });

  it('renders primary Badge', () => {
    const wrapper = shallow(<TableBadge orderStatus="Ready" />);
    expect(wrapper.props().variant).toBe('primary');
  });

  it('renders warning Badge', () => {
    const wrapper = shallow(<TableBadge orderStatus="Not Ready" />);
    expect(wrapper.props().variant).toBe('warning');
  });
});
