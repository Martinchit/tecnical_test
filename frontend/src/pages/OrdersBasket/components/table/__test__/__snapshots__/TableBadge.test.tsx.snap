// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBadge renders correctly 1`] = `
<Styled(Badge)
  variant="success"
>
  Booked
</Styled(Badge)>
`;
